common/db_test.go

```diff
@@ -656,6 +659,129 @@ func TestLoadWinningTickets_GivenTwoSessionsWithTickets_ReturnsAllTickets(t *tes
 	assert.Equal(recipientRand1, recipientRands[1])
 }
 
+func TestInsertMiniHeader_ReturnsFindLatestMiniHeader(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	assert := assert.New(t)
+	require := require.New(t)
+	require.Nil(err)
+
+	h0 := defaultMiniHeader()
+	err = dbh.InsertMiniHeader(h0)
+	require.Nil(err)
+
+	h0db, err := dbh.FindLatestMiniHeader()
+	require.Nil(err)
+	assert.Equal(*h0, *h0db)
+
+	// Test FindLatestMiniHeader with 2 blocks
+	h1 := defaultMiniHeader()
+	h1.Number = big.NewInt(451)
+	h1.Logs = append(h1.Logs, types.Log{
+		Topics:    []common.Hash{pm.RandHash(), pm.RandHash()},
+		Data:      pm.RandBytes(32),
+		BlockHash: h1.Hash,
+	})
+	err = dbh.InsertMiniHeader(h1)
+	require.Nil(err)
+	latest, err := dbh.FindLatestMiniHeader()
+	require.Nil(err)
+	assert.Equal(h1, latest)
+	assert.Equal(len(h1.Logs), 2)
+
+	// test MiniHeader = nil error
+	err = dbh.InsertMiniHeader(nil)
+	assert.EqualError(err, err.Error(), "must provide a MiniHeader")
+
+	// test blocknumber = nil error
+	h1.Number = nil
+	err = dbh.InsertMiniHeader(h1)
+	assert.EqualError(err, err.Error(), "no block number found")
+}
+
+func TestFindAllMiniHeadersSortedByNumber(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	assert := assert.New(t)
+	require := require.New(t)
+	require.Nil(err)
+
+	added := make([]*blockwatch.MiniHeader, 10)
+	for i := 0; i < 10; i++ {
+		h := defaultMiniHeader()
+		h.Number = big.NewInt(int64(i))
+		if i%2 == 0 {
+			h.Logs = append(h.Logs, types.Log{
+				Topics:    []common.Hash{pm.RandHash(), pm.RandHash()},
+				Data:      pm.RandBytes(32),
+				BlockHash: h.Hash,
+			})
+		}
+		err = dbh.InsertMiniHeader(h)
+		require.Nil(err)
+		added[9-i] = h
+	}
+
+	headers, err := dbh.FindAllMiniHeadersSortedByNumber()
+	for i, h := range headers {
+		assert.Equal(h.Number.Int64(), int64(len(headers)-1-i))
+		assert.Equal(h, added[i])
+		// Since MiniHeaders are inserted into 'added' in reverse order we need to check for uneven here instead of even
```

Comment:
This comment is a bit confusing due to the `i%2 == 0` - could be clearer to say that even index = 1 log & odd index = 2 logs

Reply:
fixed in 72b6c06